frontend/catalyst/ag_primitives.py

```diff
@@ -106,7 +107,9 @@ def functional_cond():
     # manipulate tuples when they don't expect it. Ensure set_state receives a tuple regardless.
     results = functional_cond()
     if not isinstance(results, tuple):
-        results = (results,)
+        # TODO: Do we have a test-case for this?
+        results = (results,)  # pragma: nocover
```

Comment:
Hmm, I definitely encountered that issue during development, but maybe there is no test case for it 🤔

Reply:
I think the only otrside entity that affects `result` is catalyst.for_loop. We ask it to always return `get_state` which is always a tuple, so I think it should be ok to remove the precaution. Did you do the development before the pytree support was added into catalyst ?

Reply:
If the current tests pass without it, I think we can remove them